core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java

```diff
@@ -54,6 +54,13 @@
  * creating them could perform worse than Connector. Therefore, it would be better to create an
  * AccumuloClient and pass it around.
  *
+ * <p>
+ * AccumuloClient objects are intended to be thread-safe, and can be used by multiple threads.
+ * However, care should be taken to ensure that the client is eventually closed, to clean up any
+ * resources in use in the client application when all threads are finished with the AccumuloClient
```

Comment:
This sounds great.  Now we need to document thread safety for the scanner and batch writer, if its not already done.

This information about closing overlaps a bit with information in the previous paragraph.

Reply:
> This information about closing overlaps a bit with information in the previous paragraph.

That is intentional for two reasons:
1. I wanted to have it repeated in this paragraph to expand on its brief mention in the previous paragraph, and
2. The previous paragraph may go away at some future point, and I wanted this paragraph to stand on its own.